### PR TITLE
Added migration function from xion accounts

### PR DESCRIPTION
**Pull request by Kayanski**
This PR aims at modifying the account migration function to be able to migrate xion accounts to the Abstract Account

### Checklist

- [ ] CI is green.
- [ ] Changelog updated.

### CODE REVIEW

File: framework/contracts/account/Cargo.toml
Comment (Buckram123) [inline framework/contracts/account/Cargo.toml:1]: What's going on with tabs in this file? Are you using taplo for formatting?

File: framework/contracts/account/src/migrate.rs [hunk 1]
```diff
@@ -1,19 +1,121 @@
...
-    assert_contract_upgrade(deps.storage, ACCOUNT, version)?;
+    // If we already have an abstract account, we just migrate like normal
+    if assert_contract_upgrade(deps.storage, ACCOUNT, version).is_ok() {
+        set_contract_version(deps.storage, ACCOUNT, CONTRACT_VERSION)?;
+        return Ok(AccountResponse::action("migrate"));
+    }
+    // else this means that we are migrating from another contract, we assert it's `account` and create an abstract account
+    let current_contract_version = get_contract_version(deps.storage)?;
+    if current_contract_version.contract != "account" {
+        return Err(AccountError::Std(StdError::generic_err("Can't migrate something other than `abstract_account`  or `account` to `abstract_account`")));
```
Comment (Buckram123) [inline framework/contracts/account/src/migrate.rs:43]:
```suggestion
        return Err(AccountError::Std(StdError::generic_err("Can't migrate something other than `abstract::account`  or `account` to `abstract_account`")));
```

File: framework/contracts/account/src/migrate.rs [hunk 2]
```diff
@@ -1,19 +1,121 @@
...
+    }
+
+    // Use CW2 to set the contract version, this is needed for migrations
     set_contract_version(deps.storage, ACCOUNT, CONTRACT_VERSION)?;
-    Ok(AccountResponse::action("migrate"))
+
+    let registry = RegistryContract::new(deps.api, &env)?;
```
Comment (Buckram123) [inline framework/contracts/account/src/migrate.rs:49]: Would it be possible to share a function for this and instantiate?

File: framework/packages/abstract-interface/src/deployment.rs [hunk 3]
```diff
@@ -306,6 +306,10 @@ impl<Chain: CwEnv> Abstract<Chain> {
             blob: self.blob.clone(),
         }
     }
+
+    pub fn account_code_id(&self) -> Result<u64, AbstractInterfaceError> {
```
Comment (Buckram123) [inline framework/packages/abstract-interface/src/deployment.rs:310]: Do we want code_id method on "account" object instead? Where we load code_id for "abstract::account" from cw orch env state

Comment (Buckram123) [reply framework/packages/abstract-interface/src/deployment.rs:310]: Right now it will fail because contract id is different depending on account id

**Review by Buckram123 — CHANGES_REQUESTED** [review]
Looks good, small details before we merge it

File: framework/contracts/account/Cargo.toml
Comment (Kayanski) [reply framework/contracts/account/Cargo.toml:1]: Ah oups, yes, I was experimenting with new taplo settings and it just applied. Will revert (see this task https://linear.app/abstract-sdk/issue/ABS-583/change-taplo-base-configuration)

File: framework/contracts/account/src/migrate.rs (hunk 2 above)
Comment (Kayanski) [reply framework/contracts/account/src/migrate.rs:49]: not really, I removed a lot of code that was un-necessary and all that. I might want to call instantiate directly though maybe ?

File: framework/packages/abstract-interface/src/deployment.rs (hunk 3 above)
Comment (Kayanski) [reply framework/packages/abstract-interface/src/deployment.rs:310]: Yeah that's why I put that on Abstract. I wanted a way to get the account id that should be used for creating a new account. This function might want to load the latest code id from chain, that could be better I think

File: framework/contracts/account/src/migrate.rs (hunk 2 above)
Comment (Kayanski) [reply framework/contracts/account/src/migrate.rs:49]: No that's not possible as the xion authenticator is in the way. Copying the code is the best solution here I think

File: framework/contracts/account/src/migrate.rs (hunk 1 above)
Comment (Buckram123) [reply framework/contracts/account/src/migrate.rs]: @Kayanski sorry it's not fully resolved, I edited suggestion later

**Review by Buckram123 — APPROVED** [review]
LGTM

**Pull request merged**